README.md

```diff
@@ -0,0 +1,492 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module simply load all the nox targets defined in the reference noxfile
+and make them available for the model project.
+This file can be enriched by model project specific targets.
+"""
+
+import os
+import sys
+import inspect
+
+# Insert the tools/nox folder to the python path to fetch the nox_ref_file.py content
+current_dir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+common_dir = os.path.dirname(os.path.dirname(os.path.dirname(current_dir)))
+build_ref_dir = os.path.join(common_dir, "tools", "nox")
+sys.path.insert(0, build_ref_dir)
+
+# Fetch the nox_ref_file.py content
+from nox_ref_file import *
+
+# Insert below the model project specific targets
+
+```
+
+If needed, you can add some sessions, specific to the model project, at the and of the file.
+
+This file must be initiated now because the *Nox* sessions will be used to install the python dependencies (using the ``requirements.txt`` file we will see just below) either in the development docker image, or in the ``dev_dependencies`` virtual environment.
+
+##### Initiate *requirements.txt*
+
+Like the two first files, the ``requirements.txt`` is used to setup the development environment. It should include, at least, the *pip*, *nox*, *torch* and *lightning* version you plan to use:
+
+```
+pip==24.0
+nox==2024.4.15
+torch==1.13.1+cu117
+pytorch-lightning==1.5.7
+```
+
+Note this is not the final version of this file, because you will update it at the end of the process (see [Finalize the requirements.txt](#finalize-the-requirementstxt)), just before pushing your development branch to create a pull request. As a concequence, you can add other requirements right now, but any python library you will install, using ``pip install``, during your development process, will be included in the final version of the file.
+
+##### Build the development environment
+
+You are now ready to setup your development environment, either a *Docker container* or a Python *virtual environment*.
+
+###### Docker container
+
+To build your development docker image, use the generic ``Dockerfile`` in the ``docker`` folder.
+
+The 4 following parameters must be passed to the Dockerfile, the values being picked, for the 3 last ones, from the env.yaml file previously created :
+- MODEL_PROJECT_PATH : the path ``<domain>/<use-case>/<NN architecture>`` you are working on.
+- UBUNTU_IMAGE_NAME : the base Ubuntu docker image name indicated in the ``env.yaml`` file.
```

Comment:
I'm not sure to understand. Do we need to write it explicitely or will it be automatically retrieved from the env.yaml?

Reply:
It should be written explicitely using the  --build-arg docker option, as descibed in the example below. I will modify to insist on that.